djstripe/models/account.py

```diff
@@ -103,7 +104,15 @@ def get_default_api_key(self, livemode: bool = None) -> str:
 
         if api_key:
             return api_key.secret
-        return djstripe_settings.get_default_api_key(livemode)
+
+        if not strict:
+            return djstripe_settings.get_default_api_key(livemode)
+
+        # this means that the account's key we are trying to fetch is not in the DB!
+        raise InvalidRequestError(
```

Comment:
We shouldn't raise Stripe's own errors, that is bound to lead to confusion. Add a new exception type if you must.

With that said, I'm not sure this change is useful at all.